src/backend/distributed/operations/stage_protocol.c

```diff
@@ -380,7 +380,18 @@ citus_update_table_statistics(PG_FUNCTION_ARGS)
 
 	Oid distributedTableId = PG_GETARG_OID(0);
 
-	UpdateTableStatistics(distributedTableId);
+	Relation relation = try_relation_open(distributedTableId, AccessShareLock);
+
+	if (relation != NULL)
+	{
+		UpdateTableStatistics(distributedTableId);
+		relation_close(relation, AccessShareLock);
+	}
+	else
+	{
+		ereport(WARNING, (errmsg("relation with OID %u does not exist",
```

Comment:
Why do we give a warning here? I think this can be confusing for the users, any thoughts?

Reply:
Maybe the warning can be better, but the motivation is to _let the user know_ a table from `pg_dist_partition` existed before executing `SELECT citus_update_table_statistics(logicalrelid) FROM pg_dist_partition;` and was dropped concurrently.

Reply:
I think a NOTICE might be nicer: `Relation with oid is dropped concurrently, skipping to update statistics`  or something like that?

Reply:
(Or even DEBUG1 might be fine)

Reply:
umm, if the table is dropped it doesn't need to be updated and the user probably doesn't need to know about this. Why do you think it is useful for the user?

Reply:
I suggested adding this because the user might be calling citus_update_table_statistics directly and making a typo/bug. If we say nothing the user will falsely assume the function worked. Hence, a WARNING seems right, but it should be a NOTICE at minimum.

Reply:
umm, makes sense. I think we can consider updating the message then because if I call citus_upate_table_statistics with a non-existing oid then the output is incorrect:

```
[local] talha@talha:9700-7016=# SELECT citus_update_table_statistics(21111);
NOTICE:  relation with OID 21111 was concurrently dropped, skipping
 citus_update_table_statistics
-------------------------------

(1 row)

Time: 0.665 ms
```

Reply:
Agree, would be better to say "relation with OID %d does not exist"

Reply:
Makes sense, I will go with "relation with OID %d does not exist, skipping"